adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -633,6 +633,17 @@ protected void onCommandReceived(final ProtonSender sender, final CommandContext
             msg.setCorrelationId(msg.getMessageId());
         }
 
+        String replyToId = command.getReplyToId();
+        if (!command.isOneWay()) {
+            if (command.getRequestId().startsWith("0")) {
```

Comment:
IMHO we should not make any assumptions regarding the structure of the request ID generated by the `Command` class. In the end, one of the reasons why we have the `Command` class, is to encapsulate exactly such logic. Why don't we simply *re-write* the (AMQP) message's *reply-to* property according to this logic when we create the `Command` instance from the message? We could then simply forward the result of `Command.getCommandMessage()` to the device.
We could do the reverse when we receive the command response from the device, i.e. encapsulate the restoring of the original address of the message as part of creating the `CommandResponse` instance. WDYT?

Reply:
Initially I also thought of re-writing the (AMQP) message's reply-to property according to this logic when we create the Command instance from the messages. This re-writing of reply-id is needed only for the AMQP adpater to get the `device-id`. Encoding and decoding of `reply-to` for all commands irrespective of the adapters being used will add a bit overhead (time) to the total processing time of commands. So I have scoped down the Implementation only to the AMQP adapter.

Reply:
@sophokles73 I have pushed my changes. Can you pls take a look?